packages/chart/src/XYAxis.ts

```diff
@@ -640,93 +640,127 @@ export class XYAxis extends SVGWidget {
 XYAxis.prototype.mixin(Utility.SimpleSelectionMixin);
+
+export interface XYAxis {
+    orientation(): string;
+    orientation(_: string): this;
+    pointSizeColumn(): string;
+    pointSizeColumn(_: string): this;
+    minPointSize(): number;
+    minPointSize(_: number): this;
+    maxPointSize(): number;
+    maxPointSize(_: number): this;
+    use2dSelection(): boolean;
+    use2dSelection(_: boolean): this;
+    selectionMode(): boolean;
+    selectionMode(_: boolean): this;
+    xAxisTickCount(): number;
+    xAxisTickCount(_: number): this;
+    xAxisTickFormat(): string;
+    xAxisTickFormat(_: string): this;
+    xAxisType(): string;
+    xAxisType(_: string): this;
+    xAxisTypeTimePattern(): string;
+    xAxisTypeTimePattern(_: string): this;
+    xAxisDomainLow(): number | string;
+    xAxisDomainLow(_: number | string): this;
+    xAxisDomainHigh(): number | string;
+    xAxisDomainHigh(_: number | string): this;
+    xAxisOrdinalPaddingInner(): number;
+    xAxisOrdinalPaddingInner(_: number): this;
+    xAxisOrdinalPaddingOuter(): number;
+    xAxisOrdinalPaddingOuter(_: number): this;
+    xAxisOverlapMode(): string;
+    xAxisOverlapMode(_: string): this;
+    xAxisLabelRotation(): number;
+    xAxisLabelRotation(_: number): this;
+    xAxisDomainPadding(): number;
+    xAxisDomainPadding(_: number): this;
+    xAxisGuideLines(): boolean;
+    xAxisGuideLines(_: boolean): this;
+    xAxisFocus(): boolean;
+    xAxisFocus(_: boolean): this;
+    xAxisFocusHeight(): number;
+    xAxisFocusHeight(_: number): this;
+    xAxisHidden(): boolean;
+    xAxisHidden(_: boolean): this;
+    yAxisTitle(): string;
+    yAxisTitle(_: string): this;
+    yAxisTickCount(): number;
+    yAxisTickCount(_: number): this;
+    yAxisTickFormat(): string;
+    yAxisTickFormat(_: string): this;
+    yAxisType(): string;
+    yAxisType(_: string): this;
+    yAxisTypeTimePattern(): string;
+    yAxisTypeTimePattern(_: string): this;
+    yAxisTypePowExponent(): number;
+    yAxisTypePowExponent(_: number): this;
+    yAxisTypeLogBase(): number;
+    yAxisTypeLogBase(_: number): this;
+    yAxisStacked(): boolean;
+    yAxisStacked(_: boolean): this;
+    yAxisDomainLow(): number | string;
+    yAxisDomainLow(_: number | string): this;
+    yAxisDomainHigh(): number | string;
+    yAxisDomainHigh(_: number | string): this;
+    yAxisDomainPadding(): number;
+    yAxisDomainPadding(_: number): this;
+    yAxisGuideLines(): boolean;
+    yAxisGuideLines(_: boolean): this;
+    yAxisHidden(): boolean;
+    yAxisHidden(_: boolean): this;
+    regions(): object[];
+    regions(_: object[]): this;
+    layers(): XYAxis[];
+    layers(_: XYAxis[]): this;
+    xAxisType_default(_: string);
+    yAxisType_default(_: string);
+    orientation_default(_: string);
+    xAxisHidden_default(_: boolean);
+    yAxisHidden_default(_: boolean);
+    xAxisGuideLines_default(_: boolean);
+    yAxisGuideLines_default(_: boolean);
+}
```

Comment:
Will you add the correct types at the same time?  e.g. `orientation: "horizontal"|"vertical"`

Reply:
@GordonSmith added

Reply:
`orientation` was the only one on XYAxis that needs that type of change, right?